nixos/atticd.nix

```diff
@@ -11,16 +16,19 @@ let
 
   format = pkgs.formats.toml { };
 
-  checkedConfigFile = pkgs.runCommand "checked-attic-server.toml" {
-    configFile = cfg.configFile;
-  } ''
-    cat $configFile
+  checkedConfigFile =
+    pkgs.runCommand "checked-attic-server.toml"
+      {
+        configFile = cfg.configFile;
+      }
+      ''
+        cat $configFile
 
-    export ATTIC_SERVER_TOKEN_HS256_SECRET_BASE64="dGVzdCBzZWNyZXQ="
-    export ATTIC_SERVER_DATABASE_URL="sqlite://:memory:"
-    ${cfg.package}/bin/atticd --mode check-config -f $configFile
-    cat <$configFile >$out
-  '';
+        export ATTIC_SERVER_TOKEN_HS256_SECRET_BASE64="dGVzdCBzZWNyZXQ="
+        export ATTIC_SERVER_DATABASE_URL="sqlite://:memory:"
+        ${lib.getExe cfg.package} --mode check-config -f $configFile
```

Comment:
I don't think using `getExe` makes sense here or in `serviceConfig`.

This breaks using the `attic` (or `attic-nixpkgs`) package for the service, because that package has `attic` set as `mainProgram`, which is the wrong binary to use here. So setting `services.attic.package = pkgs.attic` was working fine before this change, but is broken with it.

Since this package includes both server and client, it should be usable here IMO.

I've been using that package to avoid having to build attic twice, for server and client binaries.